napari_skimage_regionprops/_table.py

```diff
@@ -29,8 +29,17 @@ def __init__(self, layer: "napari.layers.Layer", viewer: "napari.Viewer" = None)
 
         self._view = QTableWidget()
         self._view.setEditTriggers(QTableWidget.EditTrigger.NoEditTriggers)
+        # Add 'properties' and 'features' attributes if layer is Surface
+        if isinstance(layer, napari.layers.Surface):
+            # Check if features and properties are already defined
+            if not hasattr(layer, "features"):
+                layer.features = {}
+            if not hasattr(layer, "properties"):
```

Comment:
I think you are reintroducing the problem we have during this line. The napari Surface Layer never has a properties layer. So by default if there is data attached to the surface layer (for example in the features attribute which exists by default) it looks like there is no data. I think this is causing issues when we want to look at the table. 

When I comment out these lines things work as expected but ONLY IF I load the features using the load_csv function, not when I set the features from code. I will have to investigate this tomorrow in detail though as I'm not going into any specifics ATM....

Reply:
I can reproduce what you are describing. For some reason, the code never gets to [this line](https://github.com/haesleinhuepf/napari-skimage-regionprops/blob/dd48037209158e9a32fe379049cc8095b5c719fb/napari_skimage_regionprops/_table.py#L41). The `set_contents` does something I don't fully understand yet, it seems it only cares about the `properties` attribute. I will try to mirror the features there to see if it works.  

Also at least in the latest documentation, the Surface layer does not have by default neither `properties` nor `features` (check [here](https://napari.org/stable/api/napari.layers.Surface.html)).

Reply:
I made some changes to hopefully have this fixed :)